start.sh

```diff
@@ -66,43 +78,68 @@ fi
 #create start script for X11
 echo "#!/bin/bash" > /home/chromium/xstart.sh
 
-# Only for the Pi4 since setting `display_rotate=1` doesn't work for KMS
-# rotate screen if env variable is set [normal, inverted, left or right]
-if [[ ! -z "$ROTATE_DISPLAY" ]]; then
-  echo "(sleep 3 && xrandr -o $ROTATE_DISPLAY) &" >> /home/chromium/xstart.sh
-fi
-
 # if no window size has been specified, find the framebuffer size and use that
 if [[ -z ${WINDOW_SIZE+x} ]]
   then
     export WINDOW_SIZE=$( cat /sys/class/graphics/fb0/virtual_size )
     echo "Using fullscreen: $WINDOW_SIZE"
+  else
+    echo "Custom window size set to $WINDOW_SIZE"
+fi
+
+# if no window position has been specified, use the default
+if [[ -z ${WINDOW_POSITION+x} ]]
+  then
+    export WINDOW_POSITION="0,0"
+    echo "Using default window position"
+fi
+
+# rotate screen if env variable is set [normal, inverted, left or right]
+if [[ ! -z "$ROTATE_DISPLAY" ]]; then
+  echo "(sleep 3 && xrandr -o $ROTATE_DISPLAY) &" >> /home/chromium/xstart.sh
+
+  #If the display is rotated to left or right, we need to reverse the size and position coords
+  if [[ "$ROTATE_DISPLAY" == "left" ]] || [[ "$ROTATE_DISPLAY" == "right" ]]; then
+    
+    echo "Display rotated to portait. Reversing screen coordinates"
+    
+    #window size
+    REVERSED_SIZE="$(reverse_window_coordinates $WINDOW_SIZE)"
+    WINDOW_SIZE=$REVERSED_SIZE
+    echo "Reversed window size: $WINDOW_SIZE"
+    
+    #window position, if set
+    if [[ "$WINDOW_POSITION" -ne "0,0" ]]
+    then
+      REVERSED_POSITION="$(reverse_window_coordinates $WINDOW_POSITION)"
+      export WINDOW_POSITION=$REVERSED_POSITION
```

Comment:
Not sure if overwriting the WINDOW_POSITION variable could cause issues if someone is writing logic to detect it when using block. We encountered this when overwriting KIOSK value

Reply:
thoughts here @phil-d-wilson ?